src/bootstrap/config.rs

```diff
@@ -622,6 +658,37 @@ macro_rules! define_config {
     }
 }
 
+impl<T> Merge for Option<T> {
+    fn merge(&mut self, other: Self, replace: ReplaceOpt) {
+        match replace {
+            ReplaceOpt::IgnoreDuplicate => {
+                if self.is_none() {
+                    *self = other;
+                }
+            }
+            ReplaceOpt::Override => {
+                if other.is_some() {
+                    *self = other;
+                }
+            }
+            ReplaceOpt::ErrorOnDuplicate => {
+                if other.is_some() {
+                    if self.is_some() {
+                        if cfg!(test) {
+                            panic!("overriding existing option")
+                        } else {
+                            eprintln!("overriding existing option");
+                            crate::detail_exit(2);
+                        }
+                    } else {
+                        *self = other;
+                    }
+                }
+            }
```

Comment:
not a blocker, just a suggestion:

This looks more readable to me as it has less nested blocks
```suggestion
            ReplaceOpt::ErrorOnDuplicate => match (&other, &self) {
                (Some(_), Some(_)) if cfg!(test) => panic!("overriding existing option"),
                (Some(_), Some(_)) => {
                    eprintln!("overriding existing option");
                    crate::detail_exit(2);
                }
                (Some(_), None) => *self = other,
                _ => {},
            },
```

Reply:
Hmm definitely more readable, although I chose the current approach to make the logic of handling conflicts clearer in source. Maybe that's unnecessary as it's described in the doc comment - what do you think?

Reply:
Sure, it's not a blocker so this is also fine :)